src/files.c

```diff
@@ -863,3 +865,75 @@ enum verify_result verify_FTL(bool verbose)
 
 	return success ? VERIFY_OK : VERIFY_FAILED;
 }
+/**
+ * @brief Locks a file for exclusive access.
+ *
+ * This function attempts to acquire an exclusive lock on the file
+ * associated with the given file pointer.
+ *
+ * @param fp A pointer to the FILE object representing the file to be locked.
+ * @param filename The name of the file to be locked, used for logging purposes.
+ * @return true if the lock is successfully acquired, false otherwise.
+ */
+
+bool lock_file(FILE *fp, const char *filename)
+{
+	const int fd = fileno(fp);
+	if(fd < 0)
+	{
+		log_err("Failed to get file descriptor for \"%s\": %s",
```

Comment:
Shouldn't all the errors be turned into a warning? It's nothing the users can fix and we proceed anyway.

Reply:
Yes, that's a good idea.